test/functional/services/failure_debugging.js

```diff
@@ -0,0 +1,47 @@
+import { resolve } from 'path';
+import { writeFile } from 'fs';
+import mkdirp from 'mkdirp';
+import del from 'del';
+import { promisify } from 'bluebird';
+
+const writeFileAsync = promisify(writeFile);
+const mkdirAsync = promisify(mkdirp);
+
+export async function FailureDebuggingProvider({ getService }) {
+  const screenshots = getService('screenshots');
+  const config = getService('config');
+  const lifecycle = getService('lifecycle');
+  const log = getService('log');
+  const remote = getService('remote');
+
+  await del(config.get('failureDebugging.htmlDirectory'));
+
+  async function logCurrentUrl() {
+    const currentUrl = await remote.getCurrentUrl();
+    log.info(`Current URL is: ${currentUrl}`);
+  }
+
+  async function savePageHtml(name) {
+    await mkdirAsync(config.get('failureDebugging.htmlDirectory'));
+    name = name.replace(/([^ a-zA-Z0-9/-]+)/g, '_');
```

Comment:
We should just do this in `onFailure()` and pass the correct name to `savePageHtml()` and `screenshots.takeForFailure()`, no?

Reply:
Agreed.  Fixed in most recent commit.  I also hacked up the failing test name to verify the bad chars are replaced in both the html file and the png screenshot.